CONTRIBUTING.md

```diff
@@ -0,0 +1,148 @@
+# Contributing
+
+This guide is meant to help you start contributing to the Symfony CLI by
+providing some key hints and explaining specifics related to this project.
+
+## Language choice
+
+First-time contributors could be surprised by the fact that this project is
+written in Go whereas it is highly related to the Symfony Framework which is
+written in PHP.
+
+Go has been picked because it is well suited for system development and has
+close-to-zero runtime dependencies which make releasing quite easy. This is
+ideal for a tool that is used on a wide range of platforms and potentially on
+systems where the requirements to run Symfony are not met. Go is also usually
+quite easy to apprehend for PHP developers having some similarities in their
+approach.
+
+## Setup Go
+
+Contributing to the CLI, implies that one must first setup Go locally on their
+machine. Instructions are available on the official
+[Go website](https://golang.org/dl). Just pick the latest version available: Go
+will automatically download the version currently in use in the project and
+dependencies as required.
+
+## Local setup
+
+First fork this repository and clone it in some location of your liking. Next,
+try to build and run the project:
+
+```bash
+$ go build .
+```
+
+If any error happen you must fix them before going on. If no error happen, this
+should produce a binary in the project directory. By default, this binary is
+named `symfony-cli` and suffixed with `.exe` on Windows.
+
+You should be able to run it right away:
+
+```bash
+$ ./symfony-cli version
+```
+
+The binary is self-contained: you can copy it as-is to another system and/or
+execute it without any installation process.
+
+> *Tip:* This binary can be executed from anywhere by using it's absolute path.
+> This is handy during development when you need to run it in a project
+> directory and you don't want to overwrite your system-wide Symfony CLI.
```

Comment:
Do we want to also mention running `go run main.go ...`?

Reply:
I thought about it but all the sentences that came to me looked convoluted (because most of the time it involves using the `--dir` option). therefore any suggestions are welcome :)